piqe_ocp_lib/api/ocp_exceptions.py

```diff
@@ -2,78 +2,73 @@
 
 
 class ConfigError(Exception):
-    """ Raise this Exception for configuration errors """
+    """Raise this Exception for configuration errors"""
 
     def __init__(self, msg):
         Exception.__init__(self, msg)
         self.msg = msg
 
 
-class ParseError(Exception):
-    """ Raise this Exception for file parse errors """
+class ExecutionError(Exception):
+    """Raise this Exception for command execution failures"""
 
     def __init__(self, msg):
         Exception.__init__(self, msg)
         self.msg = msg
 
 
-class ExecutionError(Exception):
-    """ Raise this Exception for command execution failures """
+class OcpException(Exception):
```

Comment:
Is it better to extend 'Exception' as opposed to ApiException?
https://github.com/kubernetes-client/python/blob/master/kubernetes/client/exceptions.py#L84
ApiException seems to extend OpenApiException which in turn extends Exception. Do you think that there might be any benefits to extending ApiException instead of the Exception? I would be interested in understanding the pros and cons. This occurred to me because it seems that the ocp specific custom exceptions all extend OcpException. Thanks

Reply:
I created the OcpException with these things in mind:

1. Pass a simple error message about all the errors related to piqe library that may occur, that includes errors from this project and ApiExceptions

2. Distinguish OcpException from ApiException, since OcpException may be not be related to ApiException always. All ApiException contain the standard http response, maybe we don't need that in all kind of errors. And we don't have to override the ApiException methods

3.  Flexibility to enhance the error message by passing extra arguments in future. Right now there is not much added there except the message, but depending on the usecase(like ocp specific error) it can be enhanced. Currently, I am fetching the extra information about errors from ApiException http response, and raising custom ocp_exceptions with message
https://github.com/piqe-test-libraries/piqe-ocp-lib/blob/db56afba1e84ab6d89f626ee17ee19db08bc64eb/piqe_ocp_lib/api/ocp_exception_handler.py#L35

Now that I am revisiting this, I am thinking if passing extra parameter to the custom exceptions is really needed. This discussion is quite helpful https://stackoverflow.com/questions/1319615/proper-way-to-declare-custom-exceptions-in-modern-python
Let me know what you think

Reply:
Regarding your question about the benefit, if we extend `ApiException` instead of the `Exception`, we can pass down the http_response's status,reason,body parameters to our `OcpException`, and use those in an error message derived from `OcpException`. 
But right now, I am getting the error message when handling `ApiException` itself. There is not many usecases yet, where I had to handle `OcpException` & print its error message.

Drawback would be as mentioned in above point 2. So far, all errors come from the API and use the http response. But I am thinking of some new  exceptions we create in piqe library which may not be of type `ApiException`

On the other hand a custom exception extending `Exception` class does not need to define/override some of those parameters. Regarding of which base class we extend, we can make the custom exception as custom we need it to be